File: clients/csharp/src/LowdownLabs.Fever/Model/MediaInfo.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using FileParameter = LowdownLabs.Fever.Client.FileParameter;
using OpenAPIDateConverter = LowdownLabs.Fever.Client.OpenAPIDateConverter;

namespace LowdownLabs.Fever.Model
{
    [DataContract(Name = "MediaInfo")]
    public partial class MediaInfo : IValidatableObject
    {
        [JsonConverter(typeof(StringEnumConverter))]
        public enum KindEnum
        {
            [EnumMember(Value = "image")]
            Image = 1,

            [EnumMember(Value = "video")]
            Video = 2,

            [EnumMember(Value = "audio")]
            Audio = 3,

            [EnumMember(Value = "pdf")]
            Pdf = 4,

            [EnumMember(Value = "doc")]
            Doc = 5,

            [EnumMember(Value = "note")]
            Note = 6
        }

        [DataMember(Name = "kind", EmitDefaultValue = true)]
        public KindEnum? Kind { get; set; }
        [JsonConstructorAttribute]
        protected MediaInfo() { }
        public MediaInfo(int blobId = default, string caption = default, DateTime? capturedAt = default, string customerId = default, string device = default, string externalRef = default, int frameCount = 0, decimal? gpsLat = default, decimal? gpsLon = default, KindEnum? kind = default, Dictionary<string, Object> metadata = default, string ocrText = default, decimal? synthetic = default, List<string> tags = default, int transcriptSegments = 0)
        {
            this.BlobId = blobId;
            this.Caption = caption;
            this.CapturedAt = capturedAt;
            this.CustomerId = customerId;
            this.Device = device;
            this.ExternalRef = externalRef;
            this.FrameCount = frameCount;
            this.GpsLat = gpsLat;
            this.GpsLon = gpsLon;
            this.Kind = kind;
            this.Metadata = metadata;
            this.OcrText = ocrText;
            this.Synthetic = synthetic;
            this.Tags = tags;
            this.TranscriptSegments = transcriptSegments;
        }

        [DataMember(Name = "blob_id", IsRequired = true, EmitDefaultValue = true)]
        public int BlobId { get; set; }

        [DataMember(Name = "caption", EmitDefaultValue = true)]
        public string Caption { get; set; }

        [DataMember(Name = "captured_at", EmitDefaultValue = true)]
        public DateTime? CapturedAt { get; set; }

        [DataMember(Name = "customer_id", EmitDefaultValue = true)]
        public string CustomerId { get; set; }

        [DataMember(Name = "device", EmitDefaultValue = true)]
        public string Device { get; set; }

        [DataMember(Name = "external_ref", EmitDefaultValue = true)]
        public string ExternalRef { get; set; }

        [DataMember(Name = "frame_count", EmitDefaultValue = false)]
        public int FrameCount { get; set; }

        [DataMember(Name = "gps_lat", EmitDefaultValue = true)]
        public decimal? GpsLat { get; set; }

        [DataMember(Name = "gps_lon", EmitDefaultValue = true)]
        public decimal? GpsLon { get; set; }

        [DataMember(Name = "metadata", EmitDefaultValue = true)]
        public Dictionary<string, Object> Metadata { get; set; }

        [DataMember(Name = "ocr_text", EmitDefaultValue = true)]
        public string OcrText { get; set; }

        [DataMember(Name = "synthetic", EmitDefaultValue = true)]
        public decimal? Synthetic { get; set; }

        [DataMember(Name = "tags", EmitDefaultValue = false)]
        public List<string> Tags { get; set; }

        [DataMember(Name = "transcript_segments", EmitDefaultValue = false)]
        public int TranscriptSegments { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class MediaInfo {\n");
            sb.Append("  BlobId: ").Append(BlobId).Append("\n");
            sb.Append("  Caption: ").Append(Caption).Append("\n");
            sb.Append("  CapturedAt: ").Append(CapturedAt).Append("\n");
            sb.Append("  CustomerId: ").Append(CustomerId).Append("\n");
            sb.Append("  Device: ").Append(Device).Append("\n");
            sb.Append("  ExternalRef: ").Append(ExternalRef).Append("\n");
            sb.Append("  FrameCount: ").Append(FrameCount).Append("\n");
            sb.Append("  GpsLat: ").Append(GpsLat).Append("\n");
            sb.Append("  GpsLon: ").Append(GpsLon).Append("\n");
            sb.Append("  Kind: ").Append(Kind).Append("\n");
            sb.Append("  Metadata: ").Append(Metadata).Append("\n");
            sb.Append("  OcrText: ").Append(OcrText).Append("\n");
            sb.Append("  Synthetic: ").Append(Synthetic).Append("\n");
            sb.Append("  Tags: ").Append(Tags).Append("\n");
            sb.Append("  TranscriptSegments: ").Append(TranscriptSegments).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        IEnumerable<ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}

File: clients/python/fever_client/models/media_info.py
from __future__ import annotations
import pprint
import re  # noqa: F401
import json

from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field, StrictFloat, StrictInt, StrictStr, field_validator
from typing import Any, ClassVar, Dict, List, Optional, Union
from typing import Optional, Set
from typing_extensions import Self
from pydantic_core import to_jsonable_python

class MediaInfo(BaseModel):
    blob_id: StrictInt
    caption: Optional[StrictStr] = Field(default=None, description="The user_text caption/keywords supplied at ingest.")
    captured_at: Optional[datetime] = Field(default=None, description="EXIF capture time.")
    customer_id: Optional[StrictStr] = None
    device: Optional[StrictStr] = Field(default=None, description="EXIF camera make/model.")
    external_ref: Optional[StrictStr] = Field(default=None, description="s3:// URI of the stored original.")
    frame_count: Optional[StrictInt] = Field(default=0, description="Sampled video frames derived from this blob.")
    gps_lat: Optional[Union[StrictFloat, StrictInt]] = None
    gps_lon: Optional[Union[StrictFloat, StrictInt]] = None
    kind: Optional[StrictStr] = Field(default=None, description="Media kind of this blob (image/video/audio/pdf/doc/note).")
    metadata: Optional[Dict[str, Any]] = None
    ocr_text: Optional[StrictStr] = Field(default=None, description="Text OCR'd out of the item (PDF pages always, photos when enabled).")
    synthetic: Optional[Union[StrictFloat, StrictInt]] = Field(default=None, description="Likely-AI score in [0,1] (FEVER_SYNTHETIC), when scored.")
    tags: Optional[List[StrictStr]] = Field(default=None, description="Client-supplied tags merged with zero-shot autotags.")
    transcript_segments: Optional[StrictInt] = Field(default=0, description="Whisper transcript segments derived from this blob.")
    __properties: ClassVar[List[str]] = ["blob_id", "caption", "captured_at", "customer_id", "device", "external_ref", "frame_count", "gps_lat", "gps_lon", "kind", "metadata", "ocr_text", "synthetic", "tags", "transcript_segments"]

    @field_validator('kind')
    def kind_validate_enum(cls, value):
        if value is None:
            return value

        if value not in set(['image', 'video', 'audio', 'pdf', 'doc', 'note']):
            raise ValueError("must be one of enum values ('image', 'video', 'audio', 'pdf', 'doc', 'note')")
        return value

    model_config = ConfigDict(
        validate_by_name=True,
        validate_by_alias=True,
        validate_assignment=True,
        protected_namespaces=(),
    )

    def to_str(self) -> str:
        return pprint.pformat(self.model_dump(by_alias=True))

    def to_json(self) -> str:
        return json.dumps(to_jsonable_python(self.to_dict()))

    @classmethod
    def from_json(cls, json_str: str) -> Optional[Self]:
        return cls.from_dict(json.loads(json_str))

    def to_dict(self) -> Dict[str, Any]:
        excluded_fields: Set[str] = set([
        ])

        _dict = self.model_dump(
            by_alias=True,
            exclude=excluded_fields,
            exclude_none=True,
        )
        if self.caption is None and "caption" in self.model_fields_set:
            _dict['caption'] = None

        if self.captured_at is None and "captured_at" in self.model_fields_set:
            _dict['captured_at'] = None

        if self.customer_id is None and "customer_id" in self.model_fields_set:
            _dict['customer_id'] = None

        if self.device is None and "device" in self.model_fields_set:
            _dict['device'] = None

        if self.external_ref is None and "external_ref" in self.model_fields_set:
            _dict['external_ref'] = None

        if self.gps_lat is None and "gps_lat" in self.model_fields_set:
            _dict['gps_lat'] = None

        if self.gps_lon is None and "gps_lon" in self.model_fields_set:
            _dict['gps_lon'] = None

        if self.kind is None and "kind" in self.model_fields_set:
            _dict['kind'] = None

        if self.metadata is None and "metadata" in self.model_fields_set:
            _dict['metadata'] = None

        if self.ocr_text is None and "ocr_text" in self.model_fields_set:
            _dict['ocr_text'] = None

        if self.synthetic is None and "synthetic" in self.model_fields_set:
            _dict['synthetic'] = None

        return _dict

    @classmethod
    def from_dict(cls, obj: Optional[Dict[str, Any]]) -> Optional[Self]:
        if obj is None:
            return None

        if not isinstance(obj, dict):
            return cls.model_validate(obj)

        _obj = cls.model_validate({
            "blob_id": obj.get("blob_id"),
            "caption": obj.get("caption"),
            "captured_at": obj.get("captured_at"),
            "customer_id": obj.get("customer_id"),
            "device": obj.get("device"),
            "external_ref": obj.get("external_ref"),
            "frame_count": obj.get("frame_count") if obj.get("frame_count") is not None else 0,
            "gps_lat": obj.get("gps_lat"),
            "gps_lon": obj.get("gps_lon"),
            "kind": obj.get("kind"),
            "metadata": obj.get("metadata"),
            "ocr_text": obj.get("ocr_text"),
            "synthetic": obj.get("synthetic"),
            "tags": obj.get("tags"),
            "transcript_segments": obj.get("transcript_segments") if obj.get("transcript_segments") is not None else 0
        })
        return _obj

